appservice/src/editScmType.ts

```diff
@@ -0,0 +1,189 @@
+
+import { NameValuePair, SiteConfigResource, SiteSourceControl } from 'azure-arm-website/lib/models';
+import { TokenCredentials, WebResource } from 'ms-rest';
+import * as opn from 'opn';
+import * as requestP from 'request-promise';
+import * as vscode from 'vscode';
+import { IAzureNode, IParsedError, parseError, UserCancelledError } from 'vscode-azureextensionui';
+import { localize } from './localize';
+import { ScmType } from './ScmType';
+import { signRequest } from './signRequest';
+import { SiteClient } from './SiteClient';
+import { uiUtils } from './utils/uiUtils';
+import { IQuickPickItemWithData } from './wizard/IQuickPickItemWithData';
+
+export async function editScmType(client: SiteClient, node: IAzureNode, outputChannel: vscode.OutputChannel): Promise<string | undefined> {
+    const config: SiteConfigResource = await client.getSiteConfig();
+    const newScmType: string = await showScmPrompt(config.scmType);
+    if (newScmType === ScmType.GitHub) {
+        if (config.scmType !== ScmType.None) {
+            // GitHub cannot be configured if there is an existing configuration source-- a limitation of Azure
+            throw new Error(localize('configurationError', 'Configuration type must be set to "None" to connect to a GitHub repository.'));
+        }
+        await connectToGitHub(node, client, outputChannel);
+    } else {
+        config.scmType = newScmType;
+        // to update one property, a complete config file must be sent
+        await client.updateConfiguration(config);
+    }
+    outputChannel.appendLine(localize('deploymentSourceUpdated,', 'Deployment source has been updated to "{0}".', newScmType));
+    // returns the updated scmType
+    return newScmType;
+}
+
+async function showScmPrompt(currentScmType: string): Promise<string> {
+    const placeHolder: string = localize('scmPrompt', 'Select a new source.');
+    const currentSource: string = localize('currentSource', '(Current source)');
+    const scmQuickPicks: vscode.QuickPickItem[] = [];
+    // generate quickPicks to not include current type
+    for (const scmQuickPick of Object.keys(ScmType)) {
+        if (scmQuickPick === currentScmType) {
+            // put the current source at the top of the list
+            scmQuickPicks.unshift({ label: scmQuickPick, description: currentSource });
+        } else {
+            scmQuickPicks.push({ label: scmQuickPick, description: '' });
+        }
+    }
+
+    const quickPick: vscode.QuickPickItem = await vscode.window.showQuickPick(scmQuickPicks, { placeHolder: placeHolder });
+    if (quickPick === undefined || quickPick.description === currentSource) {
+        // if the user clicks the current source, treat it as a cancel
+        throw new UserCancelledError();
+    } else {
+        return quickPick.label;
+    }
+}
+
+async function connectToGitHub(node: IAzureNode, client: SiteClient, outputChannel: vscode.OutputChannel): Promise<void> {
```

Comment:
I feel like all the GitHub related code could be in its own folder or at least its own file.  Seems a little odd to throw it in editScmType